### PR TITLE
Add initial set of queries

### PR DESCRIPTION
**Pull request by jkamdjou**
Will update several queries once `distinct()` is merged, but we can get the ball rolling now

### CODE REVIEW

File: queries/authentication/dkim_any_fail.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+name: "DKIM: Fail"
+type: "query"
+source: |
+	any(headers.hops, ilike(.authentication_results.dkim, "*fail*"))
```
Comment (ianthiel) [inline queries/authentication/dkim_any_fail.yml:4]: this might error on the leading tab here?

File: queries/attachments/links_free_file_host.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+name: "Attachment with links to free file sharing service"
+type: "query"
+source: |
+  filter(
+    map(attachments,
+        map(beta.binexplode(.),
+            map(filter(.scan.url.urls, .domain.root_domain in $free_file_hosts), .url)
```
Comment (ianthiel) [inline queries/attachments/links_free_file_host.yml:7]: it looks like we're extracting `.url` here, should this be pulling out `.file_name` instead?

File: queries/attachments/links_free_subdomain_host.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+name: "Attachment with links to free subdomain host"
+type: "query"
+source: |
+  filter(
+    map(attachments,
+        map(beta.binexplode(.),
+            map(filter(.scan.url.urls, .domain.root_domain in $free_subdomain_hosts), .url)
```
Comment (ianthiel) [inline queries/attachments/links_free_subdomain_host.yml:7]: same here, should we be extracting `.file_name` ultimately?

File: queries/attachments/links_url_shortener.yml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+name: "Attachment with URL shorteners"
+type: "query"
+source: |
+  filter(
+    map(attachments,
+        map(beta.binexplode(.),
+            map(filter(.scan.url.urls, .domain.root_domain in $url_shorteners), .url)
```
Comment (ianthiel) [inline queries/attachments/links_url_shortener.yml:7]: Same here, the name suggests we should be pulling out `.file_name`

File: queries/attachments/auto_executing_macros.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+name: "Attachment with auto-executing VBA macros"
```
Comment (ianthiel) [inline queries/attachments/auto_executing_macros.yml:1]: Nit: these should all be plural, so "Attachments with auto-executing VBA macros"

File: queries/headers/suspicious_recipient_pattern.yml [hunk 6]
```diff
@@ -0,0 +1,11 @@
+name: "Suspicious Recipient pattern"
```
Comment (ianthiel) [inline queries/headers/suspicious_recipient_pattern.yml:1]: Let's consider updating this to be more descriptive until Descriptions are shown in the UI, e.g. "'To' matches 'From' but not 'Return-Path'"

**Review by ianthiel — APPROVED** [review]

File: queries/headers/all_recipients_bccd.yml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+name: "All recipients BCCd"
+type: "query"
+source: |
+  type.inbound
+  and length(recipients.to) == 0
+  and length(recipients.cc) == 0
+  and length(recipients.bcc) == 1
```
Comment (rw-access) [inline queries/headers/all_recipients_bccd.yml:7]: `==` or `>=`?
or does it show up as a single "Undisclosed recipients"?

File: queries/headers/compauth_verdicts.yml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+name: "Compauth verdicts"
+type: "query"
+source: |
+  map(filter(headers.hops, .authentication_results.compauth.verdict is not null),
+    .authentication_results.compauth.verdict)
```
Comment (rw-access) [inline queries/headers/compauth_verdicts.yml:5]: slightly more compact, but up to you
```suggestion
  filter(map(headers.hops, .authentication_results.compauth.verdict, . is not null))
```

File: queries/headers/all_recipients_bccd.yml (hunk 7 above)
Comment (jkamdjou) [reply queries/headers/all_recipients_bccd.yml:7]: believe it's a single one

**Review by ianthiel — APPROVED** [review]

**Pull request merged**